controller/appcontroller.go

```diff
@@ -779,7 +779,13 @@ func (ctrl *ApplicationController) Run(ctx context.Context, statusProcessors int
 	if err != nil {
 		log.Warnf("Cannot init sharding. Error while querying clusters list from database: %v", err)
 	} else {
-		ctrl.clusterSharding.Init(clusters)
+		appItems, err := ctrl.applicationClientset.ArgoprojV1alpha1().Applications(ctrl.namespace).List(ctx, metav1.ListOptions{})
```

Comment:
I believe using `ctrl.namespace` here would only take apps in the Argo CD namespace into account, and would omit all the apps in other namespaces (if user has apps-in-any-namespace configured).

The proper way to get all the apps for the controller would be to set the namespace argument to the empty string for the applications lister if apps-in-any-namespace is enabled, and then to loop through the results and check if the app's namespace is actually allowed, building a new list off these items.

This should probably be wrapped into its own function, but sorry, I was lazy :)

Reply:
I'll create a func.

Reply:
Added a func to the latest commit to address this comment.